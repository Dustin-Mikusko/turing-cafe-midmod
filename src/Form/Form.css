form {
  display: flex;
  justify-content: space-between;
}

input {
  margin: 10px;
  height: 35px;
  width: 25%;
  font-size: 1.2em;
  padding: 5px;
}

.resy-btn {
  background-color: teal;
  color: white;
  width: 250px;
  height: 50px;
  margin-top: 10px;
  font-size: 1.2em;
}

.resy-btn:hover {
  cursor: pointer;
  background-color: rgb(3, 104, 104);
}

.resy-btn:active {
  transform: scale(.97);
}

.error {
  border: 2px solid red;
}


