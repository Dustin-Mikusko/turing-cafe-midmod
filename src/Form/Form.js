import React, { Component } from 'react';
import './Form.css';

class Form extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      date: '',
      time: '',
      guests: '',
      error: {
        name: '',
        date: '',
        time: '',
        guests: '',
      }
    }
  }


  render() {
    return (
      <form>
        <input name="name" type="text" placeholder="Name"></input>
        <input name="date" type="text" placeholder="Date (mm/dd)"></input>
        <input name="time" type="text" placeholder="Time (hh:mm am/pm)"></input>
        <input name="guests" type="number" placeholder="Number of Guests"></input>
        <button type="button" className="resy-btn">Make Reservation</button>
      </form>
    )
  }
}

export default Form;
