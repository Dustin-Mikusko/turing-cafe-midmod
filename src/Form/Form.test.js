import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  it('should render component with correct data and match snapshot', () => {
    const wrapper = shallow(<Form />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when handleChange is called', () => {
    const wrapper = shallow(<Form />);
    const mockEvent = { target: {name: 'name', value: 'Dustin'} };
    const expected = 'Dustin';

    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state('name')).toEqual(expected);
  });

  it('should update error state if any input is empty when make reservation button is clicked', () => {
    const wrapper = shallow(<Form />);
    const mockState = {
      name: 'Dustin',
      date: '7/13',
      time: '',
      guests: 4,
      error: {
        name: false,
        date: false,
        time: false,
        guests: false,
      }
    };
    const expected = {
      name: false,
      date: false,
      time: true,
      guests: false,
    }

    wrapper.setState(mockState);
    
    wrapper.find('button').simulate('click');

    expect(wrapper.state('error')).toEqual(expected);
  })
})

