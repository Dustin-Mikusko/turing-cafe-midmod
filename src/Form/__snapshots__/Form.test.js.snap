// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render component with correct data and match snapshot 1`] = `
<form>
  <input
    className=""
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className=""
    name="date"
    onChange={[Function]}
    placeholder="Date (mm/dd)"
    type="text"
    value=""
  />
  <input
    className=""
    name="time"
    onChange={[Function]}
    placeholder="Time (hh:mm am/pm)"
    type="text"
    value=""
  />
  <input
    className=""
    name="guests"
    onChange={[Function]}
    placeholder="Number of Guests"
    type="number"
    value=""
  />
  <button
    className="resy-btn"
    onClick={[Function]}
    type="button"
  >
    Make Reservation
  </button>
</form>
`;
